File: output/nemotron-vs-27b/20260326-093607/agentic/workspaces/agentic_ts_metrics_contract_patch/27b_ud_q4_xl/repeat-2/packages/protocol/src/messages.ts
export type RuntimeMetrics = {
  turnsTotal: number;
  invalidOutputCount: number;
  actionAppliedCount: number;
  actionRejectedCount: number;
  queuedActions: number;
};

export type RuntimeState = {
  model: string | null;
  tickMs: number;
  metrics: RuntimeMetrics;
};

export type SessionStateMessage = {
  type: "session.state";
  payload: {
    phase: "idle" | "running" | "paused";
    runtime: RuntimeState;
  };
};

export type WorldSnapshotMessage = {
  type: "world.snapshot";
  payload: {
    tick: number;
    world: Record<string, unknown>;
  };
};

File: output/model-compare-27b-v2/20260324-125656/agentic/workspaces/agentic_python_metrics_contract/27b_ud_q4_xl/repeat-1/protocol/messages.py
from __future__ import annotations

def make_session_state(phase: str, runtime: dict) -> dict:
    return {
        "type": "session.state",
        "payload": {
            "phase": phase,
            "runtime": runtime,
        },
    }


def make_world_snapshot(tick: int, world: dict) -> dict:
    return {
        "type": "world.snapshot",
        "payload": {
            "tick": tick,
            "world": world,
        },
    }
